src/main/java/core/basesyntax/Application.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Figure;
+
+public class Application {
+    private static final int FIGURES_COUNT = 10;
+
+    public static void main(String[] args) {
+        final FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[FIGURES_COUNT];
+
+        for (int i = 0; i < FIGURES_COUNT; i++) {
+            if (i < FIGURES_COUNT / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+        }
+
+        for (Figure figure : figures) {
+            figure.draw();

```

Comment:
No need to iterate twice, you can draw in first cycle

Reply:
Yes there are such requirement, but I don`t see a conflict. As an example: requirement is to read some filed and make all letters in upper case, in your logic you should do 2 cycles:
1. Reading file
2. Iterating by lines to make them upper-case
But this could be done in one cycle and such an approach is significantly more efficient.

Reply:
Yes there are such requirement, but I don`t see a conflict. As an example: the requirement is to read some file and make all letters in upper case, in your logic you should do 2 cycles:
1. Reading file line by line
2. Iterating by lines to make them upper-case
But this could be done in one cycle and such an approach is significantly more efficient.
So please remove the second cycle and iterate over elements only once